extension/user/login.js

```diff
@@ -0,0 +1,53 @@
+const crypto = require('crypto')
+const {EINVALIDCREDENTIALS, ENOTFOUND, customError} = require('../error')
+/**
+ * @typedef {Object} LoginInputArgs
+ * @property {string} strategy
+ * @property {Object} parameters
+ * @property {string} parameters.login
+ * @property {string} parameters.password
+ */
+
+/**
+ * @param {SDKContext} context
+ * @param {LoginInputArgs} input
+ * @param {function} cb
+ */
+module.exports = function (context, input, cb) {
+  // is logged in
+  if (context.meta.userId) {
+    return cb(null, {
```

Comment:
This is considered to be an invalid call: Ignoring the "already logged in state" results in the user theme developer not noticing some issue (either a login, when none should be done, or displaying a login form for logged in users).
There should at least be an error log, to raise awareness, so the theme can be fixed.

Reply:
This backend action is idempotent. Every time when frontend action is triggered, backend process it with same result. Not causing additional intermediate errors